src/Php81/Resources/stubs/CURLStringFile.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+// https://php.watch/versions/8.1/CURLStringFile#CURLStringFile-polyfill
+class CURLStringFile extends CURLFile
```

Comment:
I think we should wrap this declaration in an `extension_loaded('curl')` check

Reply:
alternatively, we'd need to polyfill `CURLFile` also

Reply:
well, there is no reason to define CURLStringFile if curl is not available IMO.

and btw, we cannot polyfill CURLFile if you have an older curl version without CURLFile: the core `curl` functions would not know how to handle it (here, extending `CURLFile` is precisely the hack to make `curl_*` functions do the right thing with our `CURLStringFile` when receiving it as argument).

The proper guard is probably even `class_exists(CURLFile::class, false)`, because the `symfony/polyfill-php81` package supports PHP 7.1+ but curl does not have `CURLFile` in these older versions.

Reply:
I agree. I'd add both conditions to benefit from opcache optimizations:
`if (extension_loaded('curl') && class_exists(CURLFile::class, false)) {`

Reply:
Thank you - I added the `if` block. I don't think it will prevent Composer autoloader from _trying_ to load it, but I think the condition is necessary from the polyfill to not polyfill if `curl` extension isn't enabled.